spec/models/service_provider_identity_spec.rb

```diff
@@ -181,4 +181,60 @@
       end
     end
   end
+
+  describe '#email_address_for_sharing' do
+    let!(:last_login_email_address) do
+      create(
+        :email_address,
+        email: 'last_login@email.com',
+        user: user,
+        last_sign_in_at: 1.minute.ago,
+      )
+    end
+
+    let!(:shared_email_address) do
+      create(
+        :email_address,
+        email: 'shared@email.com',
+        user: user,
+        last_sign_in_at: 1.hour.ago,
+      )
+    end
+
+    let(:identity) do
+      create(:service_provider_identity, user: user, session_uuid: SecureRandom.uuid)
+    end
+
+    context 'when email sharing feature is enabled' do
+      before do
+        allow(IdentityConfig.store).to receive(:feature_select_email_to_share_enabled)
+          .and_return(true)
+      end
+
+      context 'when an email address for sharing has been set' do
+        it 'returns the shared email' do
+          identity.email_address = shared_email_address
```

Comment:
Minor: I might expect the assignment here to happen in a `before` block so that if we added additional test cases to this context, each would have the same common behavior of "sharing has been set" without having to duplicate the code. Similar to what we're doing with feature flag enabling.

```suggestion
      context 'when an email address for sharing has been set' do
        before do
          identity.email_address = shared_email_address
        end

        it 'returns the shared email' do
```

Same comment applies for `context` block below this.